core/src/main/java/org/elasticsearch/index/engine/Segment.java

```diff
@@ -173,6 +183,11 @@ public void readFrom(StreamInput in) throws IOException {
         } else {
             segmentSort = null;
         }
+        if (in.getVersion().onOrAfter(Version.V_6_1_0) && in.readBoolean()) {
+            attributes = in.readMap(StreamInput::readString, StreamInput::readString);
+        } else {
+            attributes = null;
```

Comment:
Why not default this to `emptyMap()` so you don't need a null check in `toXContent`?